packages/aws-cdk/test/api/sdk-provider.test.ts

```diff
@@ -138,6 +149,40 @@ describe('CLI compatible credentials loading', () => {
 
     await expect(provider.forEnvironment(`${uid}some_account_#`, 'def', Mode.ForReading)).rejects.toThrow('Need to perform AWS calls');
   });
+
+  test('even when using a profile to assume another profile, STS calls goes through the proxy', async () => {
+    // Messy mocking
+    let called = false;
+    jest.mock('proxy-agent', () => {
+      class FakeAgent extends require('https').Agent {
+        public addRequest(_: any, __: any) {
+          // This error takes 6 seconds to be completely handled. It might be retries in the SDK
+          // somewhere, or something about the Node event loop. I've spent an hour trying to figure
+          // it out and I can't, and I gave up. We'll just have to live with this.
```

Comment:
maybe add a `FIXME` here? don't see it as being blocking for now but interesting nonetheless